core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -103,13 +110,18 @@ trait MetadataCache {
   def metadataVersion(): MetadataVersion
 
   def features(): FinalizedFeaturesAndEpoch
+
+  def getRandomAliveBrokerId: Option[Int]
```

Comment:
Would it make sense to add `def getControllerNode: Option[Node]` here? It might allow for less complex usage outside of this class (e.g., maybe we can avoid checking kraft/vs in some cases?)

Reply:
Returning an `Option[Node]` doesn't really solve the problem of checking the type of controller. The real problem is this: We query metadataCache 1) either for a real kraft controller (to build communication channels, for ex.) or 2) for a random broker if we have a kraft controller (to respond to metadata and describe cluster requests). 

I initially had two different functions returning either real controller id or random broker id in the case of kraft controller. This will help reduce type-checking outside this class. But Coling advised against this.

Reply:
I see, ok we can revisit this later. I suppose it would be a bit confusing if `getControllerNode()` returned something other than the Node of `getControllerId()`